packages/firestore/src/core/sync_engine.ts

```diff
@@ -1032,9 +1032,6 @@ export class MultiTabSyncEngine extends SyncEngine
 
   async applyPrimaryState(isPrimary: boolean): Promise<void> {
     if (isPrimary === true && this.isPrimary !== true) {
-      this.isPrimary = true;
```

Comment:
I notice that sync_engine.ts uses a mixture of reading from `this.isPrimary` and `this.isPrimaryClient`. Shouldn't all reads be consolidated to the latter? If so, we should fix this in a separate PR.

Reply:
`this.isPrimaryClient` is the getter in the API. `this.isPrimary` is the underlying member variable. A more common pattern is to use `this.isPrimary` and `this._isPrimary`. I will update this is in a follow-up.